network/address.go

```diff
@@ -0,0 +1,156 @@
+package network
+
+import (
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// ConnType is a type to represent what is the type of a Conn
+// It can be of different types such as TCP, TLS etc.
+type ConnType string
+
+// Address contains the ConnType and the actual network address. It is used to connect
+// to a remote host with a Conn and to listen by a Listener.
+// A network address is comprised of the IP address and the port number joined
+// by a colon.
+// For the moment, there is no IPv6 support.
+type Address string
+
+const (
+	// PlainTCP represents a vanilla TCP connection
+	PlainTCP ConnType = "tcp"
+	// TLS represents a TLS encrypted connection over TCP
+	TLS = "tls"
+	// PURB represents a PURB encryption connection over TCP
+	PURB = "purb"
+	// Local represents a channel based connection type
+	Local = "local"
+	// InvalidConnType represents a non valid connection type
+	InvalidConnType = "wrong"
+)
+
+// typeAddressSep is the separator between the type of connection and the actual
+// ip address.
+const typeAddressSep = "://"
+
+func connType(t string) ConnType {
```

Comment:
True - I saw that. I just wondered if it's the most logical thing to do. Personally I would go for an error here.